From 912d39246f676f0c67e73a4c4552b95789620da8 Mon Sep 17 00:00:00 2001
From: Elena Popa <elena.popa@nxp.com>
Date: Tue, 21 Oct 2025 13:02:31 +0300
Subject: [PATCH] imx8m: remove the ft_add_optee_node call from SPL

When U-Boot proper is skipped, the ft_system_setup() function must be
called in SPL. This function calls, in turn, the ft_add_optee_node(),
which will not execute correctly, because at that point, rom_pointer is
not initialized with the correct values.

Upstream-Status: Inappropriate [board-specific Falcon Mode support]

Signed-off-by: Elena Popa <elena.popa@nxp.com>
---
 arch/arm/mach-imx/imx8m/soc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/mach-imx/imx8m/soc.c b/arch/arm/mach-imx/imx8m/soc.c
index 41a150c2f7c..fb301fbc75a 100644
--- a/arch/arm/mach-imx/imx8m/soc.c
+++ b/arch/arm/mach-imx/imx8m/soc.c
@@ -1699,7 +1699,7 @@ usb_modify_speed:
 		               "kaslr-seed", fdt_strerror(ret));
 	}
 
-#if defined(CONFIG_ANDROID_SUPPORT) || defined(CONFIG_ANDROID_AUTO_SUPPORT)
+#if defined(CONFIG_ANDROID_SUPPORT) || defined(CONFIG_ANDROID_AUTO_SUPPORT) || defined(CONFIG_SPL_BUILD)
 	return 0;
 #else
 	return ft_add_optee_node(blob, bd);
-- 
2.34.1

